### PR TITLE
Add MethodAccessSystemGetProperty predicate

### PR DESCRIPTION
**Pull request by JLLeitschuh**
This simple predicate looks for calls to `System.getProperty` with a compile time constant.

This pattern is pretty common. As an example: https://lgtm.com/query/1737527738205752665/

This PR is a dependency for both #4388 & #4473

### CODE REVIEW

File: java/ql/src/semmle/code/java/PrintAst.ql [hunk 1]
```diff
@@ -16,5 +16,6 @@ class PrintAstConfigurationOverride extends PrintAstConfiguration {
   /**
    * TWEAK THIS PREDICATE AS NEEDED.
    */
-  override predicate shouldPrint(Element e, Location l) { super.shouldPrint(e, l) }
+  override predicate shouldPrint(Element e, Location l) { super.shouldPrint(e, l) and 
+    not l.getFile().getBaseName().matches("SystemGetPropertyCall.java") }
```
Comment (smowton) [inline java/ql/src/semmle/code/java/PrintAst.ql:20]: Accidentally committed?

File: java/ql/src/semmle/code/java/JDK.qll [hunk 2]
```diff
@@ -211,6 +211,21 @@ class MethodSystemGetProperty extends Method {
...
 }
 
+/**
+ * Any method access to a method named `getProperty` on class `java.lang.System`.
+ */
+class MethodAccessSystemGetProperty extends MethodAccess {
+  MethodAccessSystemGetProperty() { getMethod() instanceof MethodSystemGetProperty }
+
+  /**
+   * Holds true if this is a compile-time constant call for the specified `propertyName`.
```
Comment (smowton) [inline java/ql/src/semmle/code/java/JDK.qll:221]:
```suggestion
   * Holds true if this call gets a compile-time constant property `propertyName`.
```

File: java/ql/src/semmle/code/java/JDK.qll [hunk 3]
```diff
@@ -211,6 +211,21 @@ class MethodSystemGetProperty extends Method {
   }
 }
 
+/**
+ * Any method access to a method named `getProperty` on class `java.lang.System`.
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/JDK.qll:215]:
```suggestion
 * An access to a method named `getProperty` on class `java.lang.System`.
```

File: java/ql/src/semmle/code/java/JDK.qll [hunk 4]
```diff
@@ -211,6 +211,21 @@ class MethodSystemGetProperty extends Method {
...
+ */
+class MethodAccessSystemGetProperty extends MethodAccess {
+  MethodAccessSystemGetProperty() { getMethod() instanceof MethodSystemGetProperty }
+
+  /**
+   * Holds true if this is a compile-time constant call for the specified `propertyName`.
+   * Eg. `System.getProperty("user.dir")`.
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/JDK.qll:222]:
```suggestion
   * Holds if this call has a compile-time constant first argument with the value `propertyName`.
   * For example: `System.getProperty("user.dir")`.
```

File: java/ql/test/library-tests/JDK/SystemGetPropertyCall.ql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @name SystemCall
+ * @description Test the definition of System Get Property
+ */
+
+import default
```
Comment (aschackmull) [inline java/ql/test/library-tests/JDK/SystemGetPropertyCall.ql:6]:
```suggestion
import java
```
No need for file-level qldoc in qltests.

File: java/ql/src/semmle/code/java/PrintAst.ql (hunk 1 above)
Comment (JLLeitschuh) [reply java/ql/src/semmle/code/java/PrintAst.ql:20]: Nope, this is intentional. Do you want me to just update the test result?

File: java/ql/src/semmle/code/java/JDK.qll (hunk 2 above)
Comment (JLLeitschuh) [reply java/ql/src/semmle/code/java/JDK.qll]: I believe this is resolved.

File: java/ql/src/semmle/code/java/PrintAst.ql (hunk 1 above)
Comment (aschackmull) [reply java/ql/src/semmle/code/java/PrintAst.ql:20]: Yes, please update the test result instead.

Comment (JLLeitschuh) [reply java/ql/src/semmle/code/java/PrintAst.ql]: This should be resolved now.

**Comment by JLLeitschuh** [comment]
This should be ready for another review pass. 😄 

Thanks for the quick turnarround on feedback!

**Review by aschackmull — APPROVED** [review]

**Pull request merged**